// npm-pansoft-base/lib/store/index.js
'use strict';

exports.__esModule = true;

var _vue = require('vue');

var _vue2 = _interopRequireDefault(_vue);

var _vuex = require('vuex');

var _vuex2 = _interopRequireDefault(_vuex);

var _user = require('./modules/user');

var _user2 = _interopRequireDefault(_user);

var _common = require('./modules/common');

var _common2 = _interopRequireDefault(_common);

var _tags = require('./modules/tags');

var _tags2 = _interopRequireDefault(_tags);

var _logs = require('./modules/logs');

var _logs2 = _interopRequireDefault(_logs);

var _getters = require('./getters');

var _getters2 = _interopRequireDefault(_getters);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_vue2.default.use(_vuex2.default);
var store = new _vuex2.default.Store({
    modules: {
        user: _user2.default,
        common: _common2.default,
        logs: _logs2.default,
        tags: _tags2.default
    },
    getters: _getters2.default
});

exports.default = store;
