// npm-pansoft-base/lib/api/system/dept.js
'use strict';

exports.__esModule = true;
exports.getDeptTree = exports.getDept = exports.update = exports.add = exports.remove = exports.getList = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _axios = require('npm-pansoft-base/lib/router/axios');

var _axios2 = _interopRequireDefault(_axios);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getList = exports.getList = function getList(current, size, params) {
  return (0, _axios2.default)({
    url: '/api/blade-system/dept/list',
    method: 'get',
    params: _extends({}, params, {
      current: current,
      size: size
    })
  });
};
var remove = exports.remove = function remove(ids) {
  return (0, _axios2.default)({
    url: '/api/blade-system/dept/remove',
    method: 'post',
    params: {
      ids: ids
    }
  });
};

var add = exports.add = function add(row) {
  return (0, _axios2.default)({
    url: '/api/blade-system/dept/submit',
    method: 'post',
    data: row
  });
};

var update = exports.update = function update(row) {
  return (0, _axios2.default)({
    url: '/api/blade-system/dept/submit',
    method: 'post',
    data: row
  });
};

var getDept = exports.getDept = function getDept(id) {
  return (0, _axios2.default)({
    url: '/api/blade-system/dept/detail',
    method: 'get',
    params: {
      id: id
    }
  });
};
var getDeptTree = exports.getDeptTree = function getDeptTree(tenantId) {
  return (0, _axios2.default)({
    url: '/api/blade-system/dept/tree',
    method: 'get',
    params: {
      tenantId: tenantId
    }
  });
};
