// npm-pansoft-base/lib/api/system/user.js
'use strict';

exports.__esModule = true;
exports.updatePassword = exports.resetPassword = exports.getUserInfo = exports.getUser = exports.update = exports.add = exports.remove = exports.getList = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _axios = require('npm-pansoft-base/lib/router/axios');

var _axios2 = _interopRequireDefault(_axios);

var _env = require('@/config/env');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getList = exports.getList = function getList(current, size, params) {
  return (0, _axios2.default)({
    url: '/api/blade-user/list',
    method: 'get',
    params: _extends({}, params, {
      current: current,
      size: size
    })
  });
};
var remove = exports.remove = function remove(ids) {
  return (0, _axios2.default)({
    url: '/api/blade-user/remove',
    method: 'post',
    params: {
      ids: ids
    }
  });
};

var add = exports.add = function add(row) {
  return (0, _axios2.default)({
    url: '/api/blade-user/submit',
    method: 'post',
    data: row
  });
};

var update = exports.update = function update(row) {
  return (0, _axios2.default)({
    url: '/api/blade-user/update',
    method: 'post',
    data: row
  });
};

var getUser = exports.getUser = function getUser(id) {
  return (0, _axios2.default)({
    url: '/api/blade-user/detail',
    method: 'get',
    params: {
      id: id
    }
  });
};

var getUserInfo = exports.getUserInfo = function getUserInfo() {
  return (0, _axios2.default)({
    url: _env.baseUrl + '/api/blade-user/info',
    method: 'post'
  });
};

var resetPassword = exports.resetPassword = function resetPassword(userIds) {
  return (0, _axios2.default)({
    url: '/api/blade-user/reset-password',
    method: 'post',
    params: {
      userIds: userIds
    }
  });
};

var updatePassword = exports.updatePassword = function updatePassword(oldPassword, newPassword, newPassword1) {
  return (0, _axios2.default)({
    url: '/api/blade-user/update-password',
    method: 'post',
    params: {
      oldPassword: oldPassword,
      newPassword: newPassword,
      newPassword1: newPassword1
    }
  });
};
