// npm-pansoft-base/lib/api/logs.js
'use strict';

exports.__esModule = true;
exports.getErrorLogs = exports.getApiLogs = exports.getUsualLogs = exports.getErrorList = exports.getApiList = exports.getUsualList = undefined;

var _axios = require('npm-pansoft-base/lib/router/axios');

var _axios2 = _interopRequireDefault(_axios);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getUsualList = exports.getUsualList = function getUsualList(current, size) {
  return (0, _axios2.default)({
    url: '/api/blade-log/usual/list',
    method: 'get',
    params: {
      current: current,
      size: size
    }
  });
};

var getApiList = exports.getApiList = function getApiList(current, size) {
  return (0, _axios2.default)({
    url: '/api/blade-log/api/list',
    method: 'get',
    params: {
      current: current,
      size: size
    }
  });
};

var getErrorList = exports.getErrorList = function getErrorList(current, size) {
  return (0, _axios2.default)({
    url: '/api/blade-log/error/list',
    method: 'get',
    params: {
      current: current,
      size: size
    }
  });
};

var getUsualLogs = exports.getUsualLogs = function getUsualLogs(id) {
  return (0, _axios2.default)({
    url: '/api/blade-log/usual/detail',
    method: 'get',
    params: {
      id: id
    }
  });
};
var getApiLogs = exports.getApiLogs = function getApiLogs(id) {
  return (0, _axios2.default)({
    url: '/api/blade-log/api/detail',
    method: 'get',
    params: {
      id: id
    }
  });
};
var getErrorLogs = exports.getErrorLogs = function getErrorLogs(id) {
  return (0, _axios2.default)({
    url: '/api/blade-log/error/detail',
    method: 'get',
    params: {
      id: id
    }
  });
};
