// npm-pansoft-base/lib/api/user.js
'use strict';

exports.__esModule = true;
exports.getConfig = exports.logout = exports.sendLogs = exports.getTopMenu = exports.getMenu = exports.refeshToken = exports.getUserInfo = exports.getButtons = exports.loginByJwt = exports.loginByUserNameForJwt = exports.loginByUsername = undefined;

var _axios = require('npm-pansoft-base/lib/router/axios');

var _axios2 = _interopRequireDefault(_axios);

var _env = require('@/config/env');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

//post方法 使用data 
//get方法 使用params

var loginByUsername = exports.loginByUsername = function loginByUsername(domain, yhid, yhkl, type, url) {
  return (0, _axios2.default)({
    url: url,
    method: 'post',
    data: {
      domain: domain,
      yhid: yhid,
      yhkl: yhkl,
      type: type
    }
  });
};

var loginByUserNameForJwt = exports.loginByUserNameForJwt = function loginByUserNameForJwt(url, yhid, domain, yhkl, jwtState) {
  return (0, _axios2.default)({
    url: url,
    method: 'post',
    data: {
      username: yhid + '.' + domain,
      password: yhkl,
      jwt: jwtState
    }
  });
};

var loginByJwt = exports.loginByJwt = function loginByJwt(url, authorization) {
  return (0, _axios2.default)({
    url: url,
    method: 'get',
    data: {
      authorization: authorization
    }
  });
};

var getButtons = exports.getButtons = function getButtons() {
  return (0, _axios2.default)({
    url: '/api/blade-system/menu/buttons',
    method: 'get'
  });
};

var getUserInfo = exports.getUserInfo = function getUserInfo() {
  return (0, _axios2.default)({
    url: _env.baseUrl + '/user/getUserInfo',
    method: 'post'
  });
};

var refeshToken = exports.refeshToken = function refeshToken() {
  return (0, _axios2.default)({
    url: _env.baseUrl + '/user/refesh',
    method: 'post'
  });
};

//获取Menu菜单需要根据不同的数据转不动的地址
var getMenu = exports.getMenu = function getMenu(id, url) {
  return (0, _axios2.default)({
    url: _env.baseUrl + url, ///user/getMenu  sysMenu/getYwdyAndGn /sysMenu/getChildrenSysMenu /sysMenu/getAllSysMenu
    method: 'post',
    data: { fbzxtbm: id }
  });
};

var getTopMenu = exports.getTopMenu = function getTopMenu(url) {
  return (0, _axios2.default)({
    url: _env.baseUrl + url, ///user/getTopMenu  /sysMenu/getChildrenSysMenu
    method: 'post'
  });
};

var sendLogs = exports.sendLogs = function sendLogs(list) {
  return (0, _axios2.default)({
    url: _env.baseUrl + '/user/logout',
    method: 'post',
    data: list
  });
};

var logout = exports.logout = function logout() {
  return (0, _axios2.default)({
    url: '/user/logout',
    method: 'get'
  });
};
//获取配置文件
var getConfig = exports.getConfig = function getConfig() {
  return (0, _axios2.default)({
    url: _env.baseUrl + '/login/config',
    method: 'post'
  });
};
