// npm-pansoft-base/lib/store/modules/logs.js
'use strict';

exports.__esModule = true;

var _store = require('../../utils/store');

var _date = require('../../utils/date');

var logs = {
    state: {
        logsList: (0, _store.getStore)({ name: 'logsList' }) || []
    },
    actions: {
        //发送错误日志
        SendLogs: function SendLogs(_ref) {
            var state = _ref.state,
                commit = _ref.commit;

            return new Promise(function (resolve, reject) {
                //sendLogs(state.logsList).then(() => {
                commit('CLEAR_LOGS');
                resolve();
                // }).catch(error => {
                //  reject(error)
                //})
            });
        }
    },
    mutations: {
        ADD_LOGS: function ADD_LOGS(state, _ref2) {
            var type = _ref2.type,
                message = _ref2.message,
                stack = _ref2.stack,
                info = _ref2.info;

            state.logsList.push(Object.assign({
                url: window.location.href,
                time: (0, _date.dateFormat)(new Date())
            }, {
                type: type,
                message: message,
                stack: stack,
                info: info.toString()
            }));
            (0, _store.setStore)({ name: 'logsList', content: state.logsList });
        },
        CLEAR_LOGS: function CLEAR_LOGS(state) {
            state.logsList = [];
            (0, _store.setStore)({ name: 'logsList', content: state.logsList });
        }
    }

};

exports.default = logs;
