// npm-pansoft-base/lib/store/modules/tags.js
'use strict';

exports.__esModule = true;

var _store = require('../../utils/store');

var _util = require('../../utils/util');

var _website = require('@/config/website');

var _website2 = _interopRequireDefault(_website);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var isFirstPage = _website2.default.isFirstPage;
var tagWel = _website2.default.fistPage;
var tagObj = {
    label: '', //标题名称
    value: '', //标题的路径
    params: '', //标题的路径参数
    query: '', //标题的参数
    meta: {}, //额外参数
    group: [] //分组

    //处理首个标签
};function setFistTag(list) {
    if (list.length == 1) {
        list[0].close = false;
    } else {
        list.forEach(function (ele) {
            if (ele.value === tagWel.value && isFirstPage === false) {
                ele.close = false;
            } else {
                ele.close = true;
            }
        });
    }
}

var navs = {
    state: {
        tagList: (0, _store.getStore)({ name: 'tagList' }) || [],
        tag: (0, _store.getStore)({ name: 'tag' }) || tagObj,
        tagWel: tagWel
    },
    actions: {},
    mutations: {
        ADD_TAG: function ADD_TAG(state, action) {
            state.tag = action;
            (0, _store.setStore)({ name: 'tag', content: state.tag, type: 'session' });
            if (state.tagList.some(function (ele) {
                return (0, _util.diff)(ele, action);
            })) return;
            state.tagList.push(action);
            setFistTag(state.tagList);
            (0, _store.setStore)({ name: 'tagList', content: state.tagList, type: 'session' });
        },
        DEL_TAG: function DEL_TAG(state, action) {
            state.tagList = state.tagList.filter(function (item) {
                return !(0, _util.diff)(item, action);
            });
            setFistTag(state.tagList);
            (0, _store.setStore)({ name: 'tagList', content: state.tagList, type: 'session' });
        },
        DEL_ALL_TAG: function DEL_ALL_TAG(state) {
            state.tagList = [state.tagWel];
            (0, _store.setStore)({ name: 'tagList', content: state.tagList, type: 'session' });
        },
        DEL_TAG_OTHER: function DEL_TAG_OTHER(state) {
            state.tagList = state.tagList.filter(function (item) {
                if (item.value === state.tag.value) {
                    return true;
                } else if (!_website2.default.isFirstPage && item.value === _website2.default.fistPage.value) {
                    return true;
                }
            });
            setFistTag(state.tagList);
            (0, _store.setStore)({ name: 'tagList', content: state.tagList, type: 'session' });
        }
    }
};
exports.default = navs;
