// npm-pansoft-base/lib/utils/menutils.js
'use strict';

exports.__esModule = true;
exports.instMainMenus = instMainMenus;
exports.loadLeftMenus = loadLeftMenus;
exports.loadTopMenus = loadTopMenus;
exports.buildMenu = buildMenu;
exports.exchangeMenu = exchangeMenu;
exports.exchangeChildMenu = exchangeChildMenu;
exports.buildNewMenu = buildNewMenu;
exports.exchangeNewMenu = exchangeNewMenu;
exports.exchangeChildNewMenu = exchangeChildNewMenu;

var _env = require('@/config/env');

var _defaultMenuIcon = require('@/config/defaultMenuIcon');

var _defaultMenuIcon2 = _interopRequireDefault(_defaultMenuIcon);

var _util = require('./util');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function instMainMenus(isLocalMenus, menusPostion) {
    if (menusPostion == 'left') {
        return "";
    } else {
        return "";
    }
}

function loadLeftMenus(isLocalMenus) {
    var navMenuUrl = _env.baseUrl + "/sysMenu/getChildrenSysMenu?_s=" + new Date().getTime();
    if (isLocalMenus == 'Y') navMenuUrl = '/localmenus/childrensyslist.json?_s=' + new Date().getTime();
}

function loadTopMenus(isLocalMenus) {
    var navMenuUrl = _env.baseUrl + "/sysMenu/getChildrenSysMenu?_s=" + new Date().getTime();
    if (isLocalMenus == 'Y') navMenuUrl = '/localmenus/childrensyslist.json?_s=' + new Date().getTime();
}

function buildMenu(menu) {
    var newMenu_list = [];
    if (menu) {
        for (var i in menu) {
            var temp_menu = exchangeMenu(menu[i]);
            newMenu_list.push(temp_menu);
        }
    }
    //newMenu_list[newMenu_list.length-1].children.push(menuUtil);
    return newMenu_list;
}

function exchangeMenu(menu) {
    var newMenu = {};
    var defaultIcon = (0, _util.getObjFromArray)(_defaultMenuIcon2.default, { code: "id", value: "default" });
    if (menu) {
        //console.log(menu);
        newMenu.id = menu['id'];
        newMenu.parentId = menu['pid'];
        newMenu.label = menu['text'];
        var iconObj = (0, _util.getObjFromArray)(_defaultMenuIcon2.default, { code: "id", value: newMenu.id });
        if (!(0, _util.isEmpty)(iconObj)) {
            newMenu.icon = iconObj[0]['iconcls'] || ((0, _util.isEmpty)(defaultIcon) ? "el-icon-menu" : defaultIcon[0]['iconcls']);
        } else {
            newMenu.icon = menu['iconcls'] || ((0, _util.isEmpty)(defaultIcon) ? "el-icon-menu" : defaultIcon[0]['iconcls']);
        }
        newMenu.path = ""; //给个空值
        newMenu.meta = {};
        if (menu['attributes']) {
            newMenu.path = menu['attributes'].href == null ? "" : menu['attributes'].href;
        }
        if (menu['children']) {
            newMenu.children = exchangeChildMenu(menu['children']);
        }
    }
    return newMenu;
}

function exchangeChildMenu(menu) {
    var child_List = [];
    var defaultChildIcon = (0, _util.getObjFromArray)(_defaultMenuIcon2.default, { code: "id", value: "defaultChild" });
    if (menu) {
        //console.log(menu);
        for (var i in menu) {
            var newMenu = {};
            newMenu.id = menu[i]['id'];
            newMenu.parentId = menu[i]['pid'];
            newMenu.label = menu[i]['text'];
            var iconObj = (0, _util.getObjFromArray)(_defaultMenuIcon2.default, { code: "id", value: newMenu.id });
            if (!(0, _util.isEmpty)(iconObj)) {
                newMenu.icon = iconObj[0]['iconcls'] || ((0, _util.isEmpty)(defaultChildIcon) ? 'el-icon-document' : defaultChildIcon[0]['iconcls']);
            } else {
                newMenu.icon = menu[i]['iconcls'] || ((0, _util.isEmpty)(defaultChildIcon) ? 'el-icon-document' : defaultChildIcon[0]['iconcls']);
            }
            newMenu.meta = {};
            newMenu.query = { gnbm: menu[i]['id'], gnmc: menu[i]['text'] };
            newMenu.path = ""; //给个空值
            if (menu[i]['attributes']) {
                newMenu.path = menu[i]['attributes'].href == null ? "" : menu[i]['attributes'].href; //https://form.avuejs.com/
            }
            if (menu[i]['children']) {
                newMenu.children = exchangeChildMenu(menu[i]['children']);
            }
            child_List.push(newMenu);
        }
    }
    return child_List;
}

function buildNewMenu(menu) {
    var newMenu_list = [];
    if (menu) {
        for (var i in menu) {
            if (menu[i].resType == "menu") {
                var temp_menu = exchangeNewMenu(menu[i]);
                newMenu_list.push(temp_menu);
            }
        }
    }
    //newMenu_list[newMenu_list.length-1].children.push(menuUtil);
    return newMenu_list;
}

function exchangeNewMenu(menu) {
    var newMenu = {};
    var defaultIcon = (0, _util.getObjFromArray)(_defaultMenuIcon2.default, { code: "id", value: "default" });
    if (menu) {
        //console.log(menu);
        newMenu.id = menu['resourceId'];
        newMenu.parentId = menu['resParentId'];
        newMenu.label = menu['resName'];
        var iconObj = (0, _util.getObjFromArray)(_defaultMenuIcon2.default, { code: "id", value: newMenu.id });
        if (!(0, _util.isEmpty)(iconObj)) {
            newMenu.icon = iconObj[0]['iconcls'] || ((0, _util.isEmpty)(defaultIcon) ? "el-icon-menu" : defaultIcon[0]['iconcls']);
        } else {
            newMenu.icon = menu['iconcls'] || ((0, _util.isEmpty)(defaultIcon) ? "el-icon-menu" : defaultIcon[0]['iconcls']);
        }
        // newMenu.icon=menu['iconcls']||'el-icon-menu';
        newMenu.path = ""; //给个空值
        newMenu.meta = {};
        if (menu['pageUrl']) {
            newMenu.path = menu['pageUrl'] == null ? "" : menu['pageUrl'];
        }
        if (menu['children']) {
            newMenu.children = exchangeChildNewMenu(menu['children']);
        }
    }
    return newMenu;
}

function exchangeChildNewMenu(menu) {
    var child_List = [];
    var defaultChildIcon = (0, _util.getObjFromArray)(_defaultMenuIcon2.default, { code: "id", value: "defaultChild" });
    if (menu) {
        //console.log(menu);
        for (var i in menu) {
            var newMenu = {};
            newMenu.id = menu[i]['resourceId'];
            newMenu.parentId = menu[i]['resParentId'];
            newMenu.label = menu[i]['resName'];
            var iconObj = (0, _util.getObjFromArray)(_defaultMenuIcon2.default, { code: "id", value: newMenu.id });
            if (!(0, _util.isEmpty)(iconObj)) {
                newMenu.icon = iconObj[0]['iconcls'] || ((0, _util.isEmpty)(defaultChildIcon) ? 'el-icon-document' : defaultChildIcon[0]['iconcls']);
            } else {
                newMenu.icon = menu[i]['iconcls'] || ((0, _util.isEmpty)(defaultChildIcon) ? 'el-icon-document' : defaultChildIcon[0]['iconcls']);
            }
            // newMenu.icon=menu[i]['iconcls']||'el-icon-document';
            newMenu.meta = {};
            newMenu.query = { gnbm: menu[i]['resourceId'], gnmc: menu[i]['resName'] };
            newMenu.path = ""; //给个空值
            if (menu[i]['pageUrl']) {
                newMenu.path = menu[i]['pageUrl'] == null ? "" : menu[i]['pageUrl']; //https://form.avuejs.com/
            }
            if (menu[i]['children']) {
                newMenu.children = exchangeChildNewMenu(menu[i]['children']);
            }
            child_List.push(newMenu);
        }
    }
    return child_List;
}
